### PR TITLE
Update to Swift 3

### PR DESCRIPTION
**Pull request by alanzeino**
Project needed some updating to Swift 3. Updated SwiftyJSON, and migrated the rest of the code. Migrator did about 90%, manually migrated the rest.

### CODE REVIEW

**Comment by nacuteodor** [comment]
Please, also, update the executable file in the repo.

**Comment by nacuteodor** [comment]
Thanks for updating the project to Swift 3!

File: ProcessTestSummaries/main.swift [hunk 1]
```diff
@@ -299,73 +323,73 @@ func generateJUnitReport(testSummariesPlistJson testSummariesPlistJson: JSON, lo
...
+    let testSuitesTestsAttr = XMLNode.attribute(withName: "tests", stringValue: String(totalTestsCount))  as! XMLNode
+    let testSuitesFailuresAttr = XMLNode.attribute(withName: "failures", stringValue: String(totalFailuresCount)) as! XMLNode
     testSuitesNode.attributes = [testSuitesTestsAttr, testSuitesFailuresAttr]
 
     // finally, save the xml report
-    let xmlData = jUnitXml.XMLDataWithOptions(Int(NSXMLNodeOptions.NodePrettyPrint.rawValue))
-    if !xmlData.writeToFile(jUnitRepPath, atomically: false) {
-        try! CustomErrorType.InvalidArgument(error: "Writing xml data to file \(jUnitRepPath) failed!").throwsError()
+    let xmlData = jUnitXml.xmlData(withOptions: Int(XMLNode.Options.nodePrettyPrint.rawValue))
+    if !((try? xmlData.write(to: URL(fileURLWithPath: jUnitRepPath), options: [])) != nil) {
```
Comment (nacuteodor) [inline ProcessTestSummaries/main.swift]: Hey. I suppose you could check == nil, instead.

Comment (alanzeino) [reply ProcessTestSummaries/main.swift]: I think the migrator did this; I'll rewrite.

**Comment by nacuteodor** [comment]
Please, squash the commits.

**Comment by alanzeino** [comment]
Squashed commits.

**Comment by nacuteodor** [comment]
Looks good. Merging.

**Pull request merged**